File: 7kue/Find the capitals/solve-1.py
def capitals(word):
  """
  This function takes a string and returns a string with all capital letters at the beginning of each word.
  """
  capitals = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
  arr = []
  for i in range(len(word)):
    if word[i] in capitals:
      arr.append(i)
  return arr
    

print(capitals('ApplE'))  # Output: [0, 4]

File: 7kue/Homogenous arrays/solve-1.ts
export function capitals(word: string): number[] {
  let arr = []
  const isUpper = word.match(/[A-Z]/g);
  for (let i = 0; i < word.length; i++) {
    // if (word[i] === word[i].toUpperCase()) {
    if (isUpper && isUpper.includes(word[i])) {
      arr.push(i)
    }
  }
  return arr;
}

console.log(capitals("MyNameIsJohnDoe")); // Output: [ 0, 2, 6, 8, 12 ]